### ISSUE TITLE
pgsql-brute is not working

### ISSUE THREAD

**Issue by tkmru** [nmap/nmap#2606]
**Describe the bug**
pgsql-brute is not working

**To Reproduce**

```
$ docker run --name postgres-bruteforce-test -p "127.0.0.1:5432:5432" -e POSTGRES_PASSWORD=password -d postgres
$ echo 'postgres' > userdb.txt
$ echo 'password' > pass.txt
$ nmap -p 5432 --script pgsql-brute --script-args userdb=userdb.txt,passdb=pass.txt  127.0.0.1
Starting Nmap 7.93 ( https://nmap.org ) at 2023-01-22 22:16 JST
Nmap scan report for localhost (127.0.0.1)
Host is up (0.00028s latency).

PORT     STATE SERVICE
5432/tcp open  postgresql

Nmap done: 1 IP address (1 host up) scanned in 0.14 seconds
```

**Expected behavior**
An account with username `postgres` and password `password` should be discoverable through a brute force attack.

**Version info (please complete the following information):**
 - OS: macOS 12.6.2
 - Output of `nmap --version`: 7.93

**Comment by adityatelange** [comment]
If you use `-d`, it shows the error.
```
Initiating NSE at 14:18
NSE: Starting pgsql-brute against localhost (127.0.0.1:5432).
NSE: [pgsql-brute 127.0.0.1:5432] unknown auth type: 10
NSE: [pgsql-brute 127.0.0.1:5432] Trying root/ ...
NSE: [pgsql-brute 127.0.0.1:5432] unknown auth type: 10
NSE: pgsql-brute against localhost (127.0.0.1:5432) threw an error!
```

I see `unknown auth type: 10`  

which can be from:
https://github.com/nmap/nmap/blob/ca24fb11c1b40b29112c992a7aa0476cdc2d5e06/nselib/pgsql.lua#L83

or

https://github.com/nmap/nmap/blob/ca24fb11c1b40b29112c992a7aa0476cdc2d5e06/nselib/pgsql.lua#L282

Reading https://www.cybertec-postgresql.com/en/from-md5-to-scram-sha-256-in-postgresql/ I see that 
`scram-sha-256` is `v10` which matches out error given by nse.

TL;DR - NSE does not support `scram-sha-256` auth on postgres
